Automattic/pocket-casts-android
# Fix up playback service on wear

## Description
It seems that I left more than a few loose ends when I switched the watch app away from media3 to using the same playback setup as the phone app. The biggest thing I missed was updating the service declaration in wear's manifest file. This meant that the OS wasn't being informed when Pocket Casts was performing playback.

Less critically, I left the old service file and all of the DI setup for it. This PR cleans that up since we'll just switch this to media3 when we do the switch with the phone app.

> **Note**
> This PR builds on top of https://github.com/Automattic/pocket-casts-android/pull/881, so we probably want to merge that one first.

## Testing Instructions
1. Start playing an episode on a pixel watch/emulator
2. Go to the watch's home screen
3. Observe that the playback icon is active (see screencast below)

## Screenshots or Screencast 

https://user-images.githubusercontent.com/4656348/231916963-b3ffc88b-3c5f-46ac-b904-7ebb4fd27039.mov

## Checklist
